### ISSUE TITLE
Merge layers do not pass masks to the `compute_mask` function

### ISSUE THREAD

**Issue by jbed94** [keras-team/keras#19817]
After Keras & TensorFlow upgrade to 3.3 and 2.16.1 I encountered a bug with mask propagation in merge layers. A base Layer class in `__call__` function takes a first argument and treats it as a `previous_mask` ([link to source](https://github.com/keras-team/keras/blob/f6c4ac55692c132cd16211f4877fac6dbeead749/keras/src/layers/layer.py#L868C13-L875C18)). It produces an incorrect mask propagation in Merge layers where arguments are passed as a single list of tensors (and list doesn't have `_keras_mask` attribute).

Example code to reproduce unexpected behavior:

```python
from keras import layers, random, ops, Model

a = random.uniform([1, 2, 256], dtype="float32")
b = random.uniform([1, 2, 256], dtype="float32")

a._keras_mask = ops.convert_to_tensor([[True, False]])
b._keras_mask = ops.convert_to_tensor([[True, False]])

add_op = layers.Add()

x = add_op([a, b])
expected_mask = add_op.compute_mask([a, b], [a._keras_mask, b._keras_mask])

assert hasattr(x, "_keras_mask")
```

I see, that the code part is marked as `TODO`, are there any plans to fix it?

=== Edit ===

Similar code works with Keras 2.15 and TensorFlow 2.15:

```python
import tensorflow as tf

a = tf.random.uniform([1, 2, 256], dtype="float32")
b = tf.random.uniform([1, 2, 256], dtype="float32")

a._keras_mask = tf.convert_to_tensor([[True, False]])
b._keras_mask = tf.convert_to_tensor([[True, False]])

add_op = tf.keras.layers.Add()

x = add_op([a, b])
expected_mask = add_op.compute_mask([a, b], [a._keras_mask, b._keras_mask])

assert hasattr(x, "_keras_mask")
assert tf.reduce_all(x._keras_mask == expected_mask)
```

**Comment by Grvzard** [comment]
The issue has been fixed in the main branch, but it hasn't been released yet.
Related PR: #19685

**Comment by sachinprasadhs** [comment]
@jbed94 , You can use the Keras nightly till the next release, here is the working Gist using `Keras-nightly ` [![](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.sandbox.google.com/gist/sachinprasadhs/915085a4f37c054f618eac86db70c390/19817.ipynb)
